packages/app/src/elements/import-dialog.ts

```diff
@@ -65,16 +67,24 @@ export class ImportDialog extends Dialog<File, void> {
 
     renderContent() {
         const csvHasColumnsOnFirstRow = this._csvHasColumnsOnFirstRowButton?.active;
+        const hasChosenCsvSupportedFormat = Boolean(
+            this._formatSelect?.value && csvSupportedFormatValues.includes(this._formatSelect?.value)
+        );
+        const formatOptions = imp.supportedFormats.map((importFormat) => ({
+            ...importFormat,
+            disabled: hasChosenCsvSupportedFormat && !csvSupportedFormatValues.includes(importFormat.value),
```

Comment:
I ran into a bunch of problems with changing the options array (removing non-CSV options when choosing a CSV-supported format), until I found this gem was already supported in our `pl-select`. Yay!